### ISSUE TITLE
Add PSR-6 caching support.

### ISSUE THREAD

**Issue by chris001** [salesagility/SuiteCRM#3030]
#### Issue

In #3007 a new cache system Cache_Lite is proposed.  This is great for servers where no sysadmin is available to install and support binary packages of in-memory caches such as Redis, memcached, etc.

On a larger memory server it'd be preferred for the cache to use system memory for speed.

PSR-6 cache support is able to save memory caches to disk in a standardized way and survive system shutdowns and restarts.

http://www.php-fig.org/psr/psr-6/

https://github.com/symfony/cache

#### Possible Fix

1. `composer require symfony/cache`
https://packagist.org/packages/symfony/cache
https://github.com/symfony/cache
The `symfony/cache` package is maintained and updated regularly, latest commit was 3 days ago.  The SugarCache adapters haven't been updated for 3 years or more.
2. Add support in SugarCache class to use symfony\cache
3. Provide settings in `config.php` for admin to select preference for caching backend type, and hostname/port, if applicable.

#### Your Environment

* SuiteCRM Version used:
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
* Environment name and version (e.g. MySQL, PHP 7):
* Operating System and version (e.g Ubuntu 16.04):

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add PSR-6 caching support.](https://trello.com/c/qqZNYfag/34-add-psr-6-caching-support)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
